### ISSUE TITLE
Caused by com.google.crypto.tink.shaded.protobuf.InvalidProtocolBufferException:

### ISSUE THREAD

**Issue by dlwls5201** [Web3Auth/web3auth-android-sdk#46]
I follow this guide using Firebase with [Web3Auth](https://web3auth.io/docs/guides/firebase)
So It can work well however only occurs a crash happens in Galaxy S20 5G device (Android 13)

Below is my code and error log.

1. implement web3auth-android-sdk:3.3.0
```
implement org.torusresearch:web3auth-android-sdk:3.3.0
```

2. I initiated the web3auth 

```kotlin
 private val web3Auth: Web3Auth by lazy {
        val options = Web3AuthOptions(
            context = requireContext(),
            clientId = getString(R.string.web3auth_project_id),
            network = Web3Auth.Network.TESTNET,
            redirectUrl = //..
            loginConfig = hashMapOf(
                "jwt" to LoginConfigItem(
                    verifier = //..
                    typeOfLogin = TypeOfLogin.JWT,
                    name = "Firebase Login",
                    clientId = getString(R.string.web3auth_project_id)
                )
            )
        )
        Web3Auth(options)
    }
```

3. It occur crash when web3Auth calls the setResultUrl

```kotlin
    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        initView()
        initButton()
        web3Auth.setResultUrl(activity?.intent?.data) 
        activity?.addOnNewIntentListener(onNewIntentListener)
    }
```

This is a crash log. 

```
Caused by com.google.crypto.tink.shaded.protobuf.InvalidProtocolBufferException: Protocol message contained an invalid tag (zero).
       at com.google.crypto.tink.shaded.protobuf.ArrayDecoders.decodeUnknownField(ArrayDecoders.java:1036)
       at com.google.crypto.tink.shaded.protobuf.ArrayDecoders.decodeUnknownField(ArrayDecoders.java:1028)
       at com.google.crypto.tink.shaded.protobuf.ArrayDecoders.decodeUnknownField(ArrayDecoders.java:1028)
       at com.google.crypto.tink.shaded.protobuf.MessageSchema.parseProto3Message(MessageSchema.java:5426)
       at com.google.crypto.tink.shaded.protobuf.MessageSchema.mergeFrom(MessageSchema.java:5442)
       at com.google.crypto.tink.shaded.protobuf.ArrayDecoders.decodeMessageField(ArrayDecoders.java:246)
       at com.google.crypto.tink.shaded.protobuf.ArrayDecoders.decodeMessageList(ArrayDecoders.java:704)
       at com.google.crypto.tink.shaded.protobuf.MessageSchema.parseProto3Message(MessageSchema.java:5373)
       at com.google.crypto.tink.shaded.protobuf.MessageSchema.mergeFrom(MessageSchema.java:5442)
       at com.google.crypto.tink.shaded.protobuf.GeneratedMessageLite.parsePartialFrom(GeneratedMessageLite.java:1567)
       at com.google.crypto.tink.shaded.protobuf.GeneratedMessageLite.parseFrom(GeneratedMessageLite.java:1680)
       at com.google.crypto.tink.proto.Keyset.parseFrom(Keyset.java:958)
       at com.google.crypto.tink.integration.android.SharedPrefKeysetReader.read(SharedPrefKeysetReader.java:84)
       at com.google.crypto.tink.CleartextKeysetHandle.read(CleartextKeysetHandle.java:61)
       at com.google.crypto.tink.integration.android.AndroidKeysetManager$Builder.read(AndroidKeysetManager.java:332)
       at com.google.crypto.tink.integration.android.AndroidKeysetManager$Builder.readOrGenerateNewKeyset(AndroidKeysetManager.java:288)
       at com.google.crypto.tink.integration.android.AndroidKeysetManager$Builder.build(AndroidKeysetManager.java:239)
       at androidx.security.crypto.EncryptedSharedPreferences.create(EncryptedSharedPreferences.java:169)
       at com.web3auth.core.keystore.KeyStoreManagerUtils.initializePreferences(KeyStoreManagerUtils.kt:39)
       at com.web3auth.core.Web3Auth.<init>(Web3Auth.kt:72)
```

**Comment by grvgoel81** [comment]
Hey @dlwls5201, this is the issue with google library(security-crypto library) happening with android sdk versions 12 and 13 mostly. 
Issue tracker: (https://issuetracker.google.com/issues/164901843)

**Comment by grvgoel81** [comment]
Resolved in version 4.0.0
